### DISCUSSION TITLE
Google Arcore Depth API and Modelviewer integration

### DISCUSSION THREAD

**Discussion by anakin2022** [google/model-viewer#3387] [Q&A]
Hello everyone! 

 Do you know if there is already any integration or example between Google Arcore Depth API and model viewer?  to achieve the occlusion of objects... making the integration with the environment more realistic.  (Google Arcore Depth API was released in late 2019)

**Comment by elalish** [comment]
It's exposed in WebXR, but we chose not to implement it because the results are too poor / unreliable. Scene Viewer does this, so feel free to try it to see how well it works. White walls are a particular problem. Of course, things are constantly improving, so it may be time to re-evaluate.

**Reply by anakin2022** [reply]
I try it on Iphone 12 and up and the occlusion works, not for all scenes but it's on the way!

**Comment by holaworlds** [comment]
Any plan for implementing this in the near future? We need this for carpet try-out projects. thank you !!

**Reply by elalish** [reply]
Carpets are actually exactly where this fails, as the depth is imprecise compared to the placement, so the floor can often end up occluding the carpet. You're welcome to try a PR for this, in case you find a better solution.
